src/testing/erc20.rs

```diff
@@ -0,0 +1,305 @@
+#![allow(unused_imports)]
+use std::{collections::HashMap, io::Bytes, path::Path, vec};
+
+use crate::{
+    definitions::{
+        block_context::{BlockContext, StarknetChainId},
+        constants::CONSTRUCTOR_ENTRY_POINT_SELECTOR,
+    },
+    execution::{
+        execution_entry_point::ExecutionEntryPoint, CallType, TransactionExecutionContext,
+    },
+    services::api::contract_classes::deprecated_contract_class::ContractClass,
+    state::{
+        cached_state::CachedState,
+        in_memory_state_reader::InMemoryStateReader,
+        state_api::{State, StateReader},
+        ExecutionResourcesManager,
+    },
+    transaction::{error::TransactionError, DeployAccount, InvokeFunction},
+    utils::calculate_sn_keccak,
+    EntryPointType, Felt252,
+};
+use cairo_lang_starknet::casm_contract_class::CasmContractClass;
+use cairo_vm::felt::felt_str;
+use lazy_static::lazy_static;
+use num_traits::Zero;
+pub const ERC20_CONTRACT_PATH: &str = "starknet_programs/cairo2/ERC20.casm";
+use crate::{
+    state::state_cache::StorageEntry,
+    utils::{felt_to_hash, Address, ClassHash},
+};
+
+use super::{
+    get_contract_class, new_starknet_block_context_for_testing, ACCOUNT_CONTRACT_PATH, ACTUAL_FEE,
+    TEST_ACCOUNT_CONTRACT_ADDRESS, TEST_ACCOUNT_CONTRACT_CLASS_HASH, TEST_CLASS_HASH,
+    TEST_CONTRACT_ADDRESS, TEST_CONTRACT_PATH, TEST_ERC20_ACCOUNT_BALANCE_KEY,
+    TEST_ERC20_CONTRACT_CLASS_HASH,
+};
+
+pub fn call_contract<T: State + StateReader>(
+    contract_address: Felt252,
+    entrypoint_selector: Felt252,
+    calldata: Vec<Felt252>,
+    state: &mut T,
+    block_context: BlockContext,
+    caller_address: Address,
+) -> Result<Vec<Felt252>, TransactionError> {
```

Comment:
Should we update the API with this function and import it here?

Reply:
This makes sense, I totally forgot about this. I'll change this before merge.

Reply:
Pushed.